python/pyarrow/pandas_compat.py

```diff
@@ -75,7 +75,7 @@ def get_logical_type(arrow_type):
             return 'datetimetz' if arrow_type.tz is not None else 'datetime'
         elif isinstance(arrow_type, pa.lib.Decimal128Type):
             return 'decimal'
-        raise NotImplementedError(str(arrow_type))
+        return 'object'
```

Comment:
In principle we could do something similar for structs here as for lists a few lines above, eg:

```
elif isinstance(arrow_type, pa.lib.StructType):
    return 'dict[{}]'.format(... something ...)
```

but I personally wouldn't do that because 1) this is more difficult as for lists (I don't think there is currently good python api to get the names and types of the children) and 2) I don't really see the added value, as it is not a special pandas dtype anyhow from which we can use the information when reconstructing a pandas DataFrame (the same argument goes for the `"list[type]"` pandas type created for ListType, but not sure why that was added).